### ISSUE TITLE
Seeking in a replay causes a miss on a 0 miss score

### ISSUE THREAD

**Issue by Sinamer** [ppy/osu#13352]
**Describe the bug:** Watching a replay of a score I set on osu!lazer, there will always be a miss on the same slider, if you seek through the replay.

The score was set on this beatmap: https://osu.ppy.sh/beatmapsets/471285#osu/1007403

And this is the exported replay file (in zip format for github upload): 
[Sinamer playing Lindsey Stirling - Master of Tides (LunaSaika) [Violin Dance].zip](https://github.com/ppy/osu/files/6604154/Sinamer.playing.Lindsey.Stirling.-.Master.of.Tides.LunaSaika.Violin.Dance.zip)

**Screenshots or videos showing encountered issue:** 

https://user-images.githubusercontent.com/53346335/120923864-6d5dd180-c6d9-11eb-866d-bf6740d50693.mp4

**osu!lazer version:** 2021.602.0

**Logs:**
[runtime.log](https://github.com/ppy/osu/files/6604164/runtime.log)
[performance.log](https://github.com/ppy/osu/files/6604165/performance.log)

**Comment by bdach** [comment]
I'm pretty sure this isn't a new issue and has been around for a while, but I can't actually find an open issue tracking this.

**Comment by ekrctb** [comment]
I'm unable to reproduce.
I noticed the video is not showing classic mod while replay has one, and the score is different.
Can you make sure you linked the correct beatmap and replay?

**Comment by Sinamer** [comment]
This replay was done in osu!lazer, so it shouldn't have a classic mod applied. I am sure the linked beatmap and replay file is correct

**Comment by ekrctb** [comment]
It seems like current osu!lazer is always adding classic mod even for plays done in lazer. Sorry for confusion.

I could reproduce the issue by commenting out `OsuModClassic` code.

**Comment by minasminas** [comment]
This problem also persists if you don't seek in the replay. For example in this Beatmap (https://osu.ppy.sh/beatmapsets/1115124#osu/2580606 [The 4* Version])y if you Watch the Top Replay from the User "Rizer" in the Global Leaderboard, there will be at least one Miss, even though there shouldn't be any, because the Score is an SS.

**Comment by ekrctb** [comment]
@minasminas I regard this issue is specific to replays of plays done in osu!lazer. Please open another issue for issues of replay done in osu!stable.

**Comment by ekrctb** [comment]
So, the miss is in a slider head.
Without the classic mod, the slider head position is moved to follow the slider ball:
https://github.com/ppy/osu/blob/0dcdad98397453b439cab67421e29ecb3ca13f00/osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSliderHead.cs#L81

A hit circle uses the `IsHovered` property to check the hit when a button is pressed:

https://github.com/ppy/osu/blob/b82190e157deb303bf15ba69b56fb5a55ae4584d/osu.Game.Rulesets.Osu/Objects/Drawables/DrawableHitCircle.cs#L229

The issue is this `IsHovered` is only updated in `InputManager`'s update, using the `Position` from the previous frame.
The issue occurs on fast forward due to the different frame rate.
Related (?): #4288.

**Comment by integerrr** [comment]
Had a look at this again and got this on video, does this still line up with the theory ekrctb had?

https://user-images.githubusercontent.com/7279624/153291935-40f982c6-e964-40f8-9e4f-fa3426871b6a.mp4

Also see: https://github.com/ppy/osu/discussions/16833#discussioncomment-2143049

**Comment by bdach** [comment]
no, that looks like a separate different issue. note the hit error meter, the timing of the stream at ~1:08 completely changes after rewinds. also note that the previous comments were talking about sliders not hitcircles.

**Comment by ekrctb** [comment]
Only way I come up with to solve this issue is always recording *previous* frame as well as the current frame when an action is pressed. I think this is useful for increasing robustness against many frame rate-dependent situations.
(Only replays after that update are fixed).

**Comment by tybug** [comment]
Fixed by removing slider head movement in https://github.com/ppy/osu/pull/24810
